File: apps/admin_app/lib/shared/models/menu.dart
import 'category.dart';

class Menu {
  final int id;
  final int restaurantId;
  final String name;
  final String? description;
  final bool isActive;
  final int sortOrder;
  final List<Category> categories;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Menu({
    required this.id,
    required this.restaurantId,
    required this.name,
    this.description,
    this.isActive = true,
    this.sortOrder = 0,
    this.categories = const [],
    required this.createdAt,
    required this.updatedAt,
  });

  factory Menu.fromJson(Map<String, dynamic> json) => Menu(
        id: json['id'] as int,
        restaurantId: json['restaurant_id'] as int,
        name: json['name'] as String,
        description: json['description'] as String?,
        isActive: json['is_active'] as bool? ?? true,
        sortOrder: json['sort_order'] as int? ?? 0,
        categories: (json['categories'] as List<dynamic>?)
                ?.map((e) => Category.fromJson(e as Map<String, dynamic>))
                .toList() ??
            [],
        createdAt: DateTime.parse(json['created_at'] as String),
        updatedAt: DateTime.parse(json['updated_at'] as String),
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'restaurant_id': restaurantId,
        'name': name,
        'description': description,
        'is_active': isActive,
        'sort_order': sortOrder,
      };
}

File: apps/backend/app/Models/Menu.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasMany;
use Illuminate\Support\Str;

class Menu extends Model
{
    use HasFactory;

    protected static function booted(): void
    {
        static::creating(function (Menu $menu) {
            if (empty($menu->uuid)) {
                $menu->uuid = (string) Str::uuid();
            }
        });
    }

    protected $fillable = [
        'restaurant_id',
        'name',
        'description',
        'is_active',
        'sort_order',
        'uuid',
        'qr_code_path',
    ];

    protected function casts(): array
    {
        return [
            'is_active' => 'boolean',
            'sort_order' => 'integer',
        ];
    }

    public function restaurant(): BelongsTo
    {
        return $this->belongsTo(Restaurant::class);
    }

    public function categories(): HasMany
    {
        return $this->hasMany(Category::class);
    }
}
